src/ophyd_async/epics/motor.py

```diff
@@ -155,13 +157,15 @@ async def set(self, value: float, timeout: CalculatableTimeout = CALCULATE_TIMEO
             self.velocity.get_value(),
             self.acceleration_time.get_value(),
         )
-        if timeout is CALCULATE_TIMEOUT:
-            assert velocity > 0, "Motor has zero velocity"
+        if timeout is CALCULATE_TIMEOUT and velocity != 0:
             timeout = (
-                abs(new_position - old_position) / velocity
+                abs((new_position - old_position) / velocity)
                 + 2 * acceleration_time
                 + DEFAULT_TIMEOUT
             )
+        else:
```

Comment:
This else clause will fire if `timeout=1` which is not what we want...

Reply:
thanks for the catch!
fixed in 101b64ef34dad5a79988c33f4343a98205d39693
also added test cases to prevent this again.